jaggery-modules/rxt/module/scripts/asset/asset.js

```diff
@@ -577,7 +577,7 @@ var asset = {};
             if((query.hasOwnProperty(key)) && (key!='type')){
```

Comment:
There is no point of checking value of key if it does not exist in the query. AFAIK, If it does not exists, JS will short circuit and skip evaluating the value of key. 
https://en.wikipedia.org/wiki/Short-circuit_evaluation